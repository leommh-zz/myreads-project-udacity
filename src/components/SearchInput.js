import React from 'react';
import { Link } from 'react-router-dom';

const SearchInput = ({ search }) => (
    <div className="search-books">
        <div className="search-books-bar">
            <Link to="/" className="close-search"> Close </Link>
            <div className="search-books-input-wrapper">
                <input type="text" placeholder="Search by title or author" onChange={ (query) => search(query) }/>
            </div>
        </div>
        <div className="search-books-results">
            <ol className="books-grid"></ol>
        </div>
    </div>
)

export default SearchInput;