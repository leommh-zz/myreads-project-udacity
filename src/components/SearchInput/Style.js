import { nominalTypeHack } from "prop-types";

const Style = {
    main: {
        padding: 20,
        
    },
    grid: {
        margin: 20,
    },
    input: {
        backgroundColor: '#c8a3ff',
        padding: 10,
        borderRadius: 10,
        borderTop: 'none',
        borderLeft: 'none',
        borderRight: 'none',
        borderBottomWidth: 10,
        borderBottomColor: '#9d5bff', 
    },
}

export default Style;