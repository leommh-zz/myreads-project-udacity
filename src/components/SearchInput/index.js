import React from 'react';
import { Grid, Paper } from '@material-ui/core';
import {DebounceInput} from 'react-debounce-input';
import style from './style';

const SearchInput = ({ search }) => (
    <Grid item style={style.grid}>
        <Paper style={style.input} elevation={1}>
            <DebounceInput
                debounceTimeout={500}
                placeholder="Search"
                style={style.input}
                onChange={query => search(query.target.value)} 
            />
        </Paper>
    </Grid>
)

export default SearchInput;