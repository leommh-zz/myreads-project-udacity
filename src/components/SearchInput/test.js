import React from 'react';
import { shallow } from 'enzyme';
import SearchInput from './index';

describe('[Component] Search', () => {

    const search = () => console.log('Searched');
    const wrapper = shallow(<SearchInput search={search} />);

    it('Display SearchInput Component with props', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('SearchInput onChange event', () => {
        expect(wrapper.simulate('change')).toMatchSnapshot();
    });
    
});