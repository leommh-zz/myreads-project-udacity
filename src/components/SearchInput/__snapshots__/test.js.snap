// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Search Display SearchInput Component with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchInput
    search={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Paper)
        elevation={1}
        style={
                Object {
                        "backgroundColor": "#c8a3ff",
                        "borderBottomColor": "#9d5bff",
                        "borderBottomWidth": 10,
                        "borderLeft": "none",
                        "borderRadius": 10,
                        "borderRight": "none",
                        "borderTop": "none",
                        "padding": 10,
                      }
        }
>
        <DebounceInput
                debounceTimeout={500}
                element="input"
                forceNotifyByEnter={true}
                forceNotifyOnBlur={true}
                inputRef={undefined}
                minLength={0}
                onBlur={undefined}
                onChange={[Function]}
                onKeyDown={undefined}
                placeholder="Search"
                style={
                        Object {
                                "backgroundColor": "#c8a3ff",
                                "borderBottomColor": "#9d5bff",
                                "borderBottomWidth": 10,
                                "borderLeft": "none",
                                "borderRadius": 10,
                                "borderRight": "none",
                                "borderTop": "none",
                                "padding": 10,
                              }
                }
                type="text"
                value={undefined}
        />
</WithStyles(Paper)>,
      "item": true,
      "style": Object {
        "margin": 20,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <DebounceInput
          debounceTimeout={500}
          element="input"
          forceNotifyByEnter={true}
          forceNotifyOnBlur={true}
          inputRef={undefined}
          minLength={0}
          onBlur={undefined}
          onChange={[Function]}
          onKeyDown={undefined}
          placeholder="Search"
          style={
                    Object {
                              "backgroundColor": "#c8a3ff",
                              "borderBottomColor": "#9d5bff",
                              "borderBottomWidth": 10,
                              "borderLeft": "none",
                              "borderRadius": 10,
                              "borderRight": "none",
                              "borderTop": "none",
                              "padding": 10,
                            }
          }
          type="text"
          value={undefined}
/>,
        "elevation": 1,
        "style": Object {
          "backgroundColor": "#c8a3ff",
          "borderBottomColor": "#9d5bff",
          "borderBottomWidth": 10,
          "borderLeft": "none",
          "borderRadius": 10,
          "borderRight": "none",
          "borderTop": "none",
          "padding": 10,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "debounceTimeout": 500,
          "element": "input",
          "forceNotifyByEnter": true,
          "forceNotifyOnBlur": true,
          "inputRef": undefined,
          "minLength": 0,
          "onBlur": undefined,
          "onChange": [Function],
          "onKeyDown": undefined,
          "placeholder": "Search",
          "style": Object {
            "backgroundColor": "#c8a3ff",
            "borderBottomColor": "#9d5bff",
            "borderBottomWidth": 10,
            "borderLeft": "none",
            "borderRadius": 10,
            "borderRight": "none",
            "borderTop": "none",
            "padding": 10,
          },
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Paper)
          elevation={1}
          style={
                    Object {
                              "backgroundColor": "#c8a3ff",
                              "borderBottomColor": "#9d5bff",
                              "borderBottomWidth": 10,
                              "borderLeft": "none",
                              "borderRadius": 10,
                              "borderRight": "none",
                              "borderTop": "none",
                              "padding": 10,
                            }
          }
>
          <DebounceInput
                    debounceTimeout={500}
                    element="input"
                    forceNotifyByEnter={true}
                    forceNotifyOnBlur={true}
                    inputRef={undefined}
                    minLength={0}
                    onBlur={undefined}
                    onChange={[Function]}
                    onKeyDown={undefined}
                    placeholder="Search"
                    style={
                              Object {
                                        "backgroundColor": "#c8a3ff",
                                        "borderBottomColor": "#9d5bff",
                                        "borderBottomWidth": 10,
                                        "borderLeft": "none",
                                        "borderRadius": 10,
                                        "borderRight": "none",
                                        "borderTop": "none",
                                        "padding": 10,
                                      }
                    }
                    type="text"
                    value={undefined}
          />
</WithStyles(Paper)>,
        "item": true,
        "style": Object {
          "margin": 20,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <DebounceInput
            debounceTimeout={500}
            element="input"
            forceNotifyByEnter={true}
            forceNotifyOnBlur={true}
            inputRef={undefined}
            minLength={0}
            onBlur={undefined}
            onChange={[Function]}
            onKeyDown={undefined}
            placeholder="Search"
            style={
                        Object {
                                    "backgroundColor": "#c8a3ff",
                                    "borderBottomColor": "#9d5bff",
                                    "borderBottomWidth": 10,
                                    "borderLeft": "none",
                                    "borderRadius": 10,
                                    "borderRight": "none",
                                    "borderTop": "none",
                                    "padding": 10,
                                  }
            }
            type="text"
            value={undefined}
/>,
          "elevation": 1,
          "style": Object {
            "backgroundColor": "#c8a3ff",
            "borderBottomColor": "#9d5bff",
            "borderBottomWidth": 10,
            "borderLeft": "none",
            "borderRadius": 10,
            "borderRight": "none",
            "borderTop": "none",
            "padding": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "debounceTimeout": 500,
            "element": "input",
            "forceNotifyByEnter": true,
            "forceNotifyOnBlur": true,
            "inputRef": undefined,
            "minLength": 0,
            "onBlur": undefined,
            "onChange": [Function],
            "onKeyDown": undefined,
            "placeholder": "Search",
            "style": Object {
              "backgroundColor": "#c8a3ff",
              "borderBottomColor": "#9d5bff",
              "borderBottomWidth": 10,
              "borderLeft": "none",
              "borderRadius": 10,
              "borderRight": "none",
              "borderTop": "none",
              "padding": 10,
            },
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`[Component] Search SearchInput onChange event 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchInput
    search={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Paper)
        elevation={1}
        style={
                Object {
                        "backgroundColor": "#c8a3ff",
                        "borderBottomColor": "#9d5bff",
                        "borderBottomWidth": 10,
                        "borderLeft": "none",
                        "borderRadius": 10,
                        "borderRight": "none",
                        "borderTop": "none",
                        "padding": 10,
                      }
        }
>
        <DebounceInput
                debounceTimeout={500}
                element="input"
                forceNotifyByEnter={true}
                forceNotifyOnBlur={true}
                inputRef={undefined}
                minLength={0}
                onBlur={undefined}
                onChange={[Function]}
                onKeyDown={undefined}
                placeholder="Search"
                style={
                        Object {
                                "backgroundColor": "#c8a3ff",
                                "borderBottomColor": "#9d5bff",
                                "borderBottomWidth": 10,
                                "borderLeft": "none",
                                "borderRadius": 10,
                                "borderRight": "none",
                                "borderTop": "none",
                                "padding": 10,
                              }
                }
                type="text"
                value={undefined}
        />
</WithStyles(Paper)>,
      "item": true,
      "style": Object {
        "margin": 20,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <DebounceInput
          debounceTimeout={500}
          element="input"
          forceNotifyByEnter={true}
          forceNotifyOnBlur={true}
          inputRef={undefined}
          minLength={0}
          onBlur={undefined}
          onChange={[Function]}
          onKeyDown={undefined}
          placeholder="Search"
          style={
                    Object {
                              "backgroundColor": "#c8a3ff",
                              "borderBottomColor": "#9d5bff",
                              "borderBottomWidth": 10,
                              "borderLeft": "none",
                              "borderRadius": 10,
                              "borderRight": "none",
                              "borderTop": "none",
                              "padding": 10,
                            }
          }
          type="text"
          value={undefined}
/>,
        "elevation": 1,
        "style": Object {
          "backgroundColor": "#c8a3ff",
          "borderBottomColor": "#9d5bff",
          "borderBottomWidth": 10,
          "borderLeft": "none",
          "borderRadius": 10,
          "borderRight": "none",
          "borderTop": "none",
          "padding": 10,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "debounceTimeout": 500,
          "element": "input",
          "forceNotifyByEnter": true,
          "forceNotifyOnBlur": true,
          "inputRef": undefined,
          "minLength": 0,
          "onBlur": undefined,
          "onChange": [Function],
          "onKeyDown": undefined,
          "placeholder": "Search",
          "style": Object {
            "backgroundColor": "#c8a3ff",
            "borderBottomColor": "#9d5bff",
            "borderBottomWidth": 10,
            "borderLeft": "none",
            "borderRadius": 10,
            "borderRight": "none",
            "borderTop": "none",
            "padding": 10,
          },
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Paper)
          elevation={1}
          style={
                    Object {
                              "backgroundColor": "#c8a3ff",
                              "borderBottomColor": "#9d5bff",
                              "borderBottomWidth": 10,
                              "borderLeft": "none",
                              "borderRadius": 10,
                              "borderRight": "none",
                              "borderTop": "none",
                              "padding": 10,
                            }
          }
>
          <DebounceInput
                    debounceTimeout={500}
                    element="input"
                    forceNotifyByEnter={true}
                    forceNotifyOnBlur={true}
                    inputRef={undefined}
                    minLength={0}
                    onBlur={undefined}
                    onChange={[Function]}
                    onKeyDown={undefined}
                    placeholder="Search"
                    style={
                              Object {
                                        "backgroundColor": "#c8a3ff",
                                        "borderBottomColor": "#9d5bff",
                                        "borderBottomWidth": 10,
                                        "borderLeft": "none",
                                        "borderRadius": 10,
                                        "borderRight": "none",
                                        "borderTop": "none",
                                        "padding": 10,
                                      }
                    }
                    type="text"
                    value={undefined}
          />
</WithStyles(Paper)>,
        "item": true,
        "style": Object {
          "margin": 20,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <DebounceInput
            debounceTimeout={500}
            element="input"
            forceNotifyByEnter={true}
            forceNotifyOnBlur={true}
            inputRef={undefined}
            minLength={0}
            onBlur={undefined}
            onChange={[Function]}
            onKeyDown={undefined}
            placeholder="Search"
            style={
                        Object {
                                    "backgroundColor": "#c8a3ff",
                                    "borderBottomColor": "#9d5bff",
                                    "borderBottomWidth": 10,
                                    "borderLeft": "none",
                                    "borderRadius": 10,
                                    "borderRight": "none",
                                    "borderTop": "none",
                                    "padding": 10,
                                  }
            }
            type="text"
            value={undefined}
/>,
          "elevation": 1,
          "style": Object {
            "backgroundColor": "#c8a3ff",
            "borderBottomColor": "#9d5bff",
            "borderBottomWidth": 10,
            "borderLeft": "none",
            "borderRadius": 10,
            "borderRight": "none",
            "borderTop": "none",
            "padding": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "debounceTimeout": 500,
            "element": "input",
            "forceNotifyByEnter": true,
            "forceNotifyOnBlur": true,
            "inputRef": undefined,
            "minLength": 0,
            "onBlur": undefined,
            "onChange": [Function],
            "onKeyDown": undefined,
            "placeholder": "Search",
            "style": Object {
              "backgroundColor": "#c8a3ff",
              "borderBottomColor": "#9d5bff",
              "borderBottomWidth": 10,
              "borderLeft": "none",
              "borderRadius": 10,
              "borderRight": "none",
              "borderTop": "none",
              "padding": 10,
            },
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
